activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,33 @@
+*   Make `ActiveSupport::ActionableError` triggerable by existing errors.
+
+    Triggers let's you re raise an actionable error from an existing one, if it
+    matches a certain condition. Here is how Action Mailbox uses triggers to
+    raise an actionable error when the `action_mailbox_inbound_emails` table
+    does not exist.
+
+    ```ruby
+    class SetupError < Error
+      include ActiveSupport::ActionableError
+
+      def initialize(message = nil)
+        super(message || <<~MESSAGE)
+          Action Mailbox does not appear to be installed. Do you want to
+          install it now?
+        MESSAGE
+      end
+
+      trigger ActiveRecord::StatementInvalid do |error|
```

Comment:
What do you think of an api similar to the active record callbacks like: 

``` ruby
trigger on: ActiveRecord::StatementInvalid, if: ->(error) { error.to_s.match?(InboundEmail.table_name) }
```

Reply:
Yes! This interface communicates the intent better.